crates/math/src/arith_expr.rs

```diff
@@ -394,6 +394,20 @@ impl<F: Field> ArithExpr<F> {
 		}
 	}
 
+	/// Returns the number of multiplication operations in the expression.
+	///
+	/// One might care about the number of multiplications in an expression because multiplication
+	/// is a dominant operation in the evaluation of the expression in terms of computational
+	/// complexity.
+	pub fn n_mult_ops(&self) -> usize {
```

Comment:
Maybe rename to `mult_cost` since we are now making an estimate based on # of squarings too.